File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import java.util.List;
import java.util.Map;

import Bean.Result;
import com.pinyougou.pojo.TbBrand;
import page.PageResult;

/**
 * 品牌接口
 * @author Administrator
 *
 */
public interface BrandService {

	/**
	 * 从tb_brand表中查询所有的brand
	 * @return
	 */
	 List<TbBrand> findAll();

	/**
	 * 分页查询结果封装到PageBean中
	 * @param pageNum  当前页
	 * @param pageSize  当前页显示数据的条目数
	 * @return
	 */
	PageResult findByPage(int pageNum, int pageSize);

/*
	*/
/**
	 * 添加商品
	 * @param tbBrand
	 *//*

	void addBrand(TbBrand tbBrand);

	*/
/**
	 * 修改商品
	 * @param tbBrand
	 *//*

	void updateBrand(TbBrand tbBrand);
*/

	/**
	 * 添加或者更新商品并返回结果信息
	 * @param tbBrand
	 * @return
	 */
	Result addOrUpdateBrand(TbBrand tbBrand);

	/**
	 * 通过id查询商品
	 * @param id
	 * @return
	 */
	TbBrand findOneById(Long id);

	void deleteByIds(Long[] ids);

	/**
	 * 搜索查询
	 * @param tbBrand
	 * @param pageNum
	 * @param pageSize
	 * @return
	 */
    PageResult findBySearch(TbBrand tbBrand, int pageNum, int pageSize);

	/**
	 * 返回模板管理中的品牌下拉列表
	 */

	List<Map> selectOptionListForBrand();
}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
//定义brandService前端的分层模式
app.service("brandService",function ($http) {
    //添加方法绑定controller层的方法,均为和后端交互的方法
    this.findAll=function () {
        return  $http.get("../brand/findAll.do");
    };
    this.findByPage=function (pageNum,pageSize){
        return $http.get("../brand/findByPage.do?pageNum="+pageNum+"&pageSize="+pageSize);
    };
    this.findOneById=function (id) {
        return $http.get("../brand/findOneById.do?id="+id);
    };
    this.saveOrUpdateBrand=function (entity) {
        return $http.post("../brand/addOrUpdateBrand.do",entity)
    };
    this.delBrandById=function (ids) {
        return  $http.get("../brand/delete.do?ids=" +ids)
    };
    this.search=function (pageNum,pageSize,searchEntity) {
        return  $http.post("../brand/search.do?pageNum="+pageNum+"&pageSize="+pageSize,searchEntity)
    }
    this.selectOptionList=function () {
        return $http.get("../brand/selectOptionList.do");
    }


});